CodeAssignment_MVVM_Combine/Sources/Controllers/HomeViewController.swift

```diff
@@ -24,56 +25,66 @@ class HomeViewController: UIViewController {
         button.layer.borderColor = UIColor.systemGray.cgColor
         /// addActionを使うことで、objc funcを使わなくても済む
         /// UIActionのclosureを用いて関数を呼び出す
-        button.addAction(UIAction { [weak self] _ in
+        button.addAction(.init { [weak self] _ in
             self?.didTapGetAPIButton()
         }, for: .touchUpInside)
         
         return button
     }()
     
-
+    /// ViewModel
+    private let viewModel = HomeViewModel()
+    /// Cancellables
+    /// storeを利用するため : AnyCancellableを保存しておいて、当該の変数がdeinitされるとき、subscribeをキャンセルする方法
+    /// Setで複数のSubscription（購読）を１つにまとめることができ、Subscriptionの値を保持する
+    private var cancellables = Set<AnyCancellable>()
+    private var repositories: Repositories?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        setAddSubViews()
-        setupController()
-        setupConstraints()
         setupUI()
+        bind()
     }
 }
 
-/// Functions & Logics
+// MARK: - Functions & Logics
 extension HomeViewController {
-    /// Controllerをセットアップする
-    func setupController() {
+    /// ViewControllerのUIをセットアップする
+    private func setupUI() {
         view.backgroundColor = .systemBackground
-        
-        
-    }
-    
-    /// カスタムで作ったViewを全部追加する
-    func setAddSubViews() {
-        view.addSubview(getAPIButton)
+        setAddSubViews()
+        setupConstraints()
     }
     
-    /// UIをセットアップする
-    func setupUI() {
-        setupGetAPIButton()
+    /// ViewModelなどViewController側で常に監視しておくべき対象を、セットアップ
+    /// イベント発生時に正常にデータバインドをするために、Observerを設定する感じ
+    /// Combineで流れたきたデータのアウトプットsinkする
+    /// sink : Publisherからのイベントを購読する.  つまり、イベントを受信したときの処理を指定できる。
+    /// receive(on:)：イベントを受け取るスレッドを指定する
+    /// store:
```

Comment:
これは見やすくするためのtipsなんだけど、`///` のdoc-commentをつける理由は主にoptionを押しながらfuncや変数をクリックしたときのdescriptionのためにあります。
現状のこのコメントの書き方だと、↓のような感じになるんだけど
<img width="580" alt="スクリーンショット 2024-05-09 10 34 23" src="https://github.com/KyusokLee/CodeAssignment_MVVM_Combine/assets/39939005/feaa58d1-fc59-4d55-a76f-288d1cfcb053">

ここを提案のように書き換えると、こんな感じに表示が綺麗になります！
<img width="567" alt="スクリーンショット 2024-05-09 10 38 32" src="https://github.com/KyusokLee/CodeAssignment_MVVM_Combine/assets/39939005/85014be0-2b51-4924-adb3-ecaf89a3c251">

```suggestion
    /** ViewModelなどViewController側で常に監視しておくべき対象を、セットアップする関数。イベント発生時に正常にデータバインドをするために、Observerを設定する感じ

    - `sink:` Publisherからのイベントを購読する.  つまり、イベントを受信したときの処理を指定できる。
    - `receive(on:):` イベントを受け取るスレッドを指定する
    - `store:` hogehoge:
    */
```

ここらへんの見やすくするテクニックはまとめてくれてる人がいるのでチェックしてみるとよさそう。
https://qiita.com/ashdik/items/0019b6ba5ed228d41e66

Reply:
@Ogijun2018 
フィードバックありがとうございます！コード修正します！今後活用させていただきます❤️